pkg/rid/application/subscription.go

```diff
@@ -38,26 +39,44 @@ type SubscriptionApp interface {
 	SearchSubscriptionsByOwner(ctx context.Context, cells s2.CellUnion, owner dssmodels.Owner) ([]*ridmodels.Subscription, error)
 }
 
-// InsertSubscription implements the App InsertSubscription method
 func (a *app) InsertSubscription(ctx context.Context, s *ridmodels.Subscription) (*ridmodels.Subscription, error) {
+	// Validate and perhaps correct StartTime and EndTime.
+	if err := s.AdjustTimeRange(a.clock.Now(), nil); err != nil {
+		return nil, err
+	}
+
+	// Check the user hasn't created too many subscriptions in this area.
+	count, err := a.Transactor.MaxSubscriptionCountInCellsByOwner(ctx, s.Cells, s.Owner)
+	if err != nil {
+		a.logger.Error("Error fetching max subscription count", zap.Error(err))
+		return nil, dsserr.Internal(
+			"failed to fetch subscription count, rejecting request")
+	}
+	if count >= maxSubscriptionsPerArea {
+		return nil, dsserr.Exhausted(
+			"too many existing subscriptions in this area already")
+	}
+
+	sub, err := a.Transactor.InsertSubscription(ctx, s)
+	if err, ok := err.(*pq.Error); ok && err.Code == "23505" {
```

Comment:
`err.Code == "23505"` seems cryptic (as does pq.Error); does the Subscription interface document this error behavior?  We want to be implementing to [the Subscription interface](https://github.com/interuss/dss/blob/0b2d4099d8d2fc713c8adf750ea278b095cad6bb/pkg/rid/application/application.go#L17) at this layer, not a particular implementation (e.g., Cockroach).